### PR TITLE
Fix for beefy

### PR DESCRIPTION
**Pull request by yrong**
sno-516

### CODE REVIEW

File: relayer/relays/parachain/beefy-listener.go [hunk 1]
```diff
@@ -234,9 +234,8 @@ func (li *BeefyListener) generateProof(ctx context.Context, input *ProofInput, h
 
 	// Generate the MMR proof for the polkadot block.
 	mmrProof, err := li.relaychainConn.GenerateProofForBlock(
-		input.RelayBlockNumber+1,
+		input.RelayBlockNumber,
```
Comment (vgeddes) [inline relayer/relays/parachain/beefy-listener.go:237]: Ah, nope, I explicitly had to use "+ 1" here in the previous PR (refactor OutboundQueue) to get the right proofs. This is different from the beefy-relayer.

You can test BridgeHub->Ethereum messaging by calling the `EthereumControl.upgrade` extrinsic in the parachain .

File: relayer/relays/beefy/polkadot-listener.go [hunk 2]
```diff
@@ -106,15 +113,12 @@ func (li *PolkadotListener) scanCommitments(
 				case <-ctx.Done():
 					return ctx.Err()
 				case requests <- task:
+					logEntry.Info("New commitment with handover added to channel")
+					currentValidatorSet++
 				}
-			} else if result.SignedCommitment.Commitment.ValidatorSetID == currentValidatorSet {
+			} else if result.SignedCommitment.Commitment.ValidatorSetID == currentValidatorSet && result.SignedCommitment.Commitment.ValidatorSetID == uint64(result.MMRProof.Leaf.BeefyNextAuthoritySet.ID) {
```
Comment (doubledup) [inline relayer/relays/beefy/polkadot-listener.go:119]: If we have a handover at `currentValidatorSet == setNumber`, we have
```
currentValidatorSet == setNumber
Commitment.ValidatorSetID == setNumber + 1
BeefyNextAuthoritySet.ID == setNumber + 2
```

Then `currentValidatorSet` increments and we have a non-handover case where all of the ids are expected to be equal:
```
currentValidatorSet == setNumber + 1
Commitment.ValidatorSetID == setNumber + 1
BeefyNextAuthoritySet.ID == setNumber + 1
```

So `BeefyNextAuthoritySet.ID` decreases after a handover. I expected this instead:
```
currentValidatorSet == setNumber + 1
Commitment.ValidatorSetID == setNumber + 1
BeefyNextAuthoritySet.ID == setNumber + 2
```

Is `BeefyNextAuthoritySet.ID` the id after `Commitment.ValidatorSetID` (the handover case), or the id expected in the next commitment (the non-handover case)? It looks like these cases interpret `BeefyNextAuthoritySet.ID` differently.

**Review by doubledup — APPROVED** [review]
The BEEFY relayer's not blocking! 🙌 

Have you been able to test the non-handover case? I can see the handover message coming through regularly, but there are also a lot of `Discarded invalid commitment` messages and no non-handover messages (`New commitment added to channel`). This is after running the `EthereumControl.upgrade` extrinsic Vincent mentioned.

**Comment by yrong** [comment]
> Have you been able to test the non-handover case? ... but there are also a lot of Discarded invalid commitment messages and no non-handover messages (New commitment added to channel)...

@doubledup Thanks for pointing it out! Also notice that and fix in https://github.com/Snowfork/snowbridge/pull/863/commits/1c67cb4f37e1e2fcf50db014429af78a1e44d203

File: relayer/relays/beefy/polkadot-listener.go (hunk 2 above)
Comment (yrong) [reply relayer/relays/beefy/polkadot-listener.go]: There are 2 cases. Handover with `currentValidatorSet +1 = Commitment.ValidatorSetID = MMRProof.Leaf.BeefyNextAuthoritySet.ID -1` and  non-handover with `(currentValidatorSet | currentValidatorSet + 1) = Commitment.ValidatorSetID = MMRProof.Leaf.BeefyNextAuthoritySet.ID`

**Review by alistair-singh — APPROVED** [review]

**Review by vgeddes — APPROVED** [review]
Looks good!

**Pull request merged**